src/controllers/EvidenceQuestionController.ts

```diff
@@ -43,14 +43,14 @@ const schema: Joi.AnySchema = Joi.object({
 @provide(NavigationPermissionProcessor)
 class NavigationPermissionProcessor implements FormActionProcessor {
     process(request: RequestWithNavigation): void {
-        const session = request.session.unsafeCoerce();
-        const applicationData: ApplicationData = session.getExtraData()
-            .map<ApplicationData>(data => data[APPLICATION_DATA_KEY])
-            .orDefaultLazy(() => {
-                const value = {} as ApplicationData;
-                session.saveExtraData(APPLICATION_DATA_KEY, value);
-                return value;
-            });
+        const session: Session | undefined = request.session;
+
+        let applicationData: ApplicationData | undefined = session?.getExtraData(APPLICATION_DATA_KEY);
+
+        if (!applicationData) {
+            applicationData = {} as ApplicationData;
+            session!.setExtraData(APPLICATION_DATA_KEY, applicationData);
+        }
 
```

Comment:
this could be simplified
```suggestion
                    session.setExtraData(APPLICATION_DATA_KEY, applicationData || {} as ApplicationData);
```

Reply:
If applicationData is defined, we dont need to setExtraData